ckan/logic/action/create.py

```diff
@@ -306,25 +307,32 @@ def resource_create(context: Context,
     try:
         update_context = Context(context)
         update_context['original_package'] = original_package
-        updated_pkg_dict = _get_action('package_update')(
-            update_context, pkg_dict)
+        _get_action('package_update')(update_context, pkg_dict)
     except ValidationError as e:
         try:
             error_dict = cast("list[ErrorDict]", e.error_dict['resources'])[-1]
         except (KeyError, IndexError):
             error_dict = e.error_dict
         raise ValidationError(error_dict)
 
-    resource = updated_pkg_dict['resources'][-1]
+    #  Run package show again to get out actual last_resource
+    both_data = _get_action('package_show')(
+        logic.fresh_context(context, ignore_auth=True, use_cache=False),
+        {'id': package_id, 'use_default_schema': 'both'}
+    )
+    # update index because of defer_commit use above
+    logic.index_update_package_dict(both_data)
```

Comment:
What's the specific relation between defer_commit and updating the index?

Reply:
Thanks for catching this. Removed because it was no longer required.